src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -80,11 +87,12 @@ public static void main(String[] args) {
   private final Options opts;
   private final Compiler compiler;
   private final GentsErrorManager errorManager;
+  private final Map<String, String> externsMap;
 
   final PathUtil pathUtil;
   final NameUtil nameUtil;
 
-  TypeScriptGenerator(Options opts) {
+  TypeScriptGenerator(Options opts) throws FileNotFoundException {
```

Comment:
Yes. The idea would be to isolate the TypeScriptGenerator from stuff like file reading and be done with that before it runs. That avoids doing work that can fail in the constructor, and also makes sure that all inputs are validated properly before the program runs.